CommonTools/PileupAlgos/plugins/PuppiProducer.cc

```diff
@@ -365,7 +369,13 @@ void PuppiProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
       }
       // Protect high pT photons (important for gamma to hadronic recoil balance)
       if ((fPtMaxPhotons > 0) && (lPack->pdgId() == 22) && (std::abs(lPack->eta()) < fEtaMaxPhotons) &&
```

Comment:
ouldn't these two more conveniently (and understandably) get wrtitten together as:
```C++
      if ((fApplyPhotonProtectionForExistingWeights || fuseBugFix) &&
          (fPtMaxPhotons > 0) && (lPack->pdgId() == 22) &&
          (std::abs(lPack->eta()) < fEtaMaxPhotons) && (lPack->pt() > fPtMaxPhotons))
        curpupweight = 1;
```
Still this corresponds to the past behavior only if either one of `fApplyPhotonProtectionForExistingWeights` and `fuseBugFix` are `true`, which does not seem to me the wanted behaviour: am I wrong?

Reply:
We will remove the use of `fuseBugFix` in `PuppiProducer` as it is redundant in this code.